searchguard/rolesmapping.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/python3
+
+import requests
+import json
+import searchguard.settings as settings
+from searchguard.exceptions import *
+from searchguard.roles import check_role_exists
+
+
+def __process_data(role, properties):
```

Comment:
I agree some functions are not easy to name, `__process_data` seems to be too generic. May I suggest we use something more specific about what it's about to do, for example `_send_api_request`?